sn_node_manager/src/cmd/node.rs

```diff
@@ -849,3 +860,110 @@ fn summarise_any_failed_ops(
     }
     Ok(())
 }
+
+/// Check if the mount point has enough space to add the required number of nodes. It also factors in the space used
+/// by other nodes running on the same mount point.
+fn check_available_space(
```

Comment:
Hey, I would have a suggestion here.

Perhaps you could split this function out possibly into two, and have one that passes in the nodes to operate on, without passing in the registry?

Then you could add some unit tests for the function that does the storage calculation?

I think a few test cases would probably help to understand the different cases and failures here.